gymnasium_robotics/envs/multiagent_mujoco/mujoco_multi.py

```diff
@@ -0,0 +1,636 @@
+    gym_env = MaMuJoCo('Ant', '8x1', agent_factorization=my_agent_factorization)
+    ```
+    """
+
+    metadata = {
+        "render_modes": ["human", "rgb_array", "depth_array"],
+        "name": "MaMuJoCo",
+        "is_parallelizable": True,
+        # "render_fps": 0,  #depends on underlying Envrioment
+        "has_manual_policy": False,
+    }
+
+    def __init__(
+        self,
+        scenario: str,
+        agent_conf: str,
+        agent_obsk: int = 1,
+        agent_factorization: dict[str, any] = None,
+        local_categories: list[list[str]] = None,
+        global_categories: list[str] = None,
+        render_mode: str = None,
+    ):
+        """
+        Args:
+            scenario:
+                The Task/Environment, valid values:
+                "Ant", "HalfCheetah", "Hopper", "HumanoidStandup", "Humanoid", "Reacher", "Swimmer", "Walker2d", "InvertedPendulum", "InvertedDoublePendulum", "manyagent_swimmer", "manyagent_ant", "coupled_half_cheetah"
+            agent_conf:
+                '${Number Of Agents}x${Number Of Segments per Agent}${Optionally Additional options}', eg '1x6', '2x4', '2x4d',
+                if it set to None the task becomes single agent (the agent observes the entire environment, and performs all the actions)
+            agent_obsk:
+                Number of nearest joints to observe,
+                if set to 0 it only observes local state,
+                if set to 1 it observes local state + 1 joint over,
+                if set to 2 it observes local state + 2 joints over,
+                if it set to None the task becomes single agent (the agent observes the entire environment, and performs all the actions)
+                The Default value is: 1
+            agent_factorization:
+                A custom factorization of the MuJoCo environment (overwrites agent_conf),
+                see DOC [how to create new agent factorizations](link).
+            local_categories:
+                The categories of local observations for each observation depth,
+                The default is: Everything is observable at depth 0, but only the position items are observable for further depth levels
+            global_categories:
+                The categories of observation for global observations,
+                The default is; local_categories[0]
+            render_mode: see [Gymansium/MuJoCo](https://gymnasium.farama.org/environments/mujoco/),
+                valid values: 'human', 'rgb_array', 'depth_array'
+
+            Raises:
+                NotImplementedError:
+                    When the scenario is not supported (not part of of the valid values)
+        """
+        scenario += "-v4"
+
+        # load the underlying single agent Gymansium MuJoCo Environment in `self.gym_env`
+        if scenario in _MUJOCO_GYM_ENVIROMENTS:
+            self.gym_env = gymnasium.make(scenario, render_mode=render_mode)
```

Comment:
Personally, `gym_env` should be changed to `single_agent_env` as this is more clean the difference between `self` (multi-agent env) and `gym_env`